default/methods/source_control/file/file_browser.py

```diff
@@ -575,55 +612,58 @@ def __build_media_type_value(self) -> str:
-            self.log['error']['file_view_mode'] = 'Invalid file_view_mode "{}"'.format(self.metadata['file_view_mode'])
-            return None
-        ignore_id_list = None
+    def get_query_file_list(self):
+        working_dir_file_list, count = self.build_and_execute_query(
+            limit = self.metadata["limit"],
+            offset = self.metadata["start_index"],
+        )
+        if not working_dir_file_list or regular_log.log_has_error(self.log):
+            return None, None
 
-        # For creating / viewing Jobs
+        return working_dir_file_list, count
+    
+    def get_job_file_list(self):
+        
+        # TODO permissions check on job id?
 
-        if self.metadata['file_view_mode'] == "task" and self.metadata['job_id']:
+        # TODO handling for larger file sizes
 
-            # TODO permissions check on job id?
+        # TODO better null handling here... a lot of assumptions
+        # Would prefer to declare "new" job condition
+        # Instead of inferring from file_view_mode and job_id being present
 
-            # TODO handling for larger file sizes
+        job = Job.get_by_id(session = self.session,
+                            job_id = self.metadata['job_id'])
 
-            # TODO better null handling here... a lot of assumptions
-            # Would prefer to declare "new" job condition
-            # Instead of inferring from file_view_mode and job_id being present
+        file_list_attached_to_job = WorkingDirFileLink.file_list(
+            session = self.session,
+            working_dir_id = job.directory_id,
+            limit = None)
 
-            job = Job.get_by_id(session = self.session,
-                                job_id = self.metadata['job_id'])
+        # TODO future maybe just get ids only from sql
+        ignore_id_list = [i.id for i in file_list_attached_to_job]
+        output_job_file_list = []
 
-            file_list_attached_to_job = WorkingDirFileLink.file_list(
-                session = self.session,
-                working_dir_id = job.directory_id,
-                limit = None)
+        for index_file_attach, file in enumerate(file_list_attached_to_job):
+            file_serialized = file.serialize_with_type(session = self.session)
+            output_job_file_list.append(file_serialized)
+            output_job_file_list[index_file_attach]['attached_to_job'] = True
 
-            # TODO future maybe just get ids only from sql
-            ignore_id_list = [i.id for i in file_list_attached_to_job]
+        return output_job_file_list, ignore_id_list
 
-            for index_file_attach, file in enumerate(file_list_attached_to_job):
-                file_serialized = file.serialize_with_type(session = self.session)
-                output_file_list.append(file_serialized)
-                output_file_list[index_file_attach]['attached_to_job'] = True
 
+    def get_all_file_list(self, ignore_id_list = None):
```

Comment:
`get_all_file_list` replaces [this block](https://github.com/diffgram/diffgram/blob/38d12c685c47a53061bcbe9b80c1ae257e651a84/default/methods/source_control/file/file_browser.py#L627-L665) & [this block](https://github.com/diffgram/diffgram/blob/38d12c685c47a53061bcbe9b80c1ae257e651a84/default/methods/source_control/file/file_browser.py#L678-L717) of code